src/stream/node/mod.rs

```diff
@@ -196,9 +198,15 @@ where
                         .unwrap(),
                 ));
             }
-            hardware_metric_group
-                .register_hardware_metric_gauges(descriptor.clone(), perf_events)
-                .ok();
+            // hardware_metric_group
+            //     .register_hardware_metric_gauges(descriptor.clone(), perf_events)
+            //     .ok();
+            let iterator = perf_events.counters.iter();
+            for value in iterator {
+                register_histogram!(value.to_string(),"node" => descriptor.clone());
+            }
+            hardware_metric_group.group.enable().ok();
```

Comment:
should we be enabling the group here in the constructor?

Well to check that there are no errors yes, but it will start counting.  Its probably fine to have it there, but then we should probably move the group.reset()?; before rather than after executing handle_events(...).

```rust
#[cfg(feature = "hardware_counters", ... )]
self.hardware_metric_group.reset()?;
 #[cfg(feature = "metrics")]
let start_time = Instant::now();
```